### PR TITLE
✨ PPD-224: Adding new endpoint to retrieve user roles (Nomis/Auth/Delius)

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt [hunk 1]
```diff
@@ -96,6 +97,31 @@ class UserController(private val userService: UserService) {
       .map { ResponseEntity.ok(it) }.orElse(notFoundResponse(username))
   }
 
+  @GetMapping("/api/user/{username}/roles")
+  @ApiOperation(
+    value = "List of roles for user.",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt:102]: would be good to document what roles are required in the api docs too

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt [hunk 2]
```diff
@@ -96,6 +97,31 @@ class UserController(private val userService: UserService) {
...
+    notes = "List of roles for user.",
+    nickname = "userRoles",
+    consumes = "application/json",
+    produces = "application/json"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(code = 200, message = "OK", response = UserRole::class, responseContainer = "List"),
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt:110]: don't need the 200 response here, will be generated automatically

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt:110]: although would be good to document the 404

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt (hunk 1 above)
Comment (brightonsbox) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt]: 👍

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**